pub/lib/dependabot/pub/update_checker.rb

```diff
@@ -78,6 +78,31 @@ def report
       def current_report
         report.find { |d| d["name"] == dependency.name }
       end
+
+      def resolved_requirements_update_strategy
+        raise "Unexpected requirements_update_strategy #{requirements_update_strategy}" unless
+          [nil, "widen_ranges", "bump_versions", "bump_versions_if_necessary"].include? requirements_update_strategy
+
+        if requirements_update_strategy.nil? 
+          # Check for a version field in the pubspec.yaml. If it is present
+          # we assume the package is a library, and the requirement update 
+          # strategy is widening. Otherwise we assume it is an application, and
+          # go for "bump_versions".
+          pubspec = dependency_files.find { |d| d.name == "pubspec.yaml" }
+          begin
+            parsed_pubspec = YAML.safe_load(pubspec.content, aliases: false)
+          rescue ScriptError
```

Comment:
I can't find any references to `YAML.safe_load` raising `ScriptError`, but either way it might be safer to rescue `StandardError` here, do we every want this to prevent updates running with `bump_versions` if the yaml loading fails?

Reply:
I went by the explanation here https://stackoverflow.com/a/14676819/4809566. SyntaxError doesn't inherit from StandardError...

I don't know the ruby yaml parser well enough, so I thought it would be better to just have a default case here... maybe we should just fail

Reply:
Aha, interesting TIL! Yeah maybe we _should_ actually fail because in this scenario the YAML is invalid and there's probably not much we can do. It would be good to raise a `Dependabot::DependencyFileNotParseable` error in this case though, so that our internal systems know how to handle the job error correctly.

Fwiw, in other places we do:

```ruby
rescue Psych::SyntaxError, Psych::DisallowedClass, Psych::BadAlias
```